framework/web/Response.php

```diff
@@ -583,7 +588,9 @@ public function sendStreamAsFile($handle, $attachmentName, $options = [])
         if (isset($options['fileSize'])) {
             $fileSize = $options['fileSize'];
         } else {
-            fseek($handle, 0, SEEK_END);
+            if ($this->isSeekable($handle)) {
+                fseek($handle, 0, SEEK_END);
+            }
             $fileSize = ftell($handle);
```

Comment:
This line should be moved inside the conditional block above because we cannot guarantee the cursor position otherwise. And I guess default value of `$fileSize` should be then 0.

Reply:
@yiisoft/reviewers thoughts?

Reply:
I agree with @bizley. Otherwise for non-seekable streams it will give you weird unexpected result based on reading progress.